modules/multilanguage/src/main/scala/com/ukg/surge/multilanguage/GenericAsyncAggregateCommandModel.scala

```diff
@@ -42,53 +43,60 @@ class GenericAsyncAggregateCommandModel(bridgeToBusinessApp: BusinessLogicServic
-      val processCommandRequest = ProcessCommandRequest(aggregateId = surgeCommand.aggregateId, maybePbState, Some(pbCommand))
-      val reply: Future[ProcessCommandReply] = bridgeToBusinessApp.processCommand(processCommandRequest)
+  override def processCommand(aggregate: Option[SurgeState], surgeCommand: SurgeCmd): Future[Seq[SurgeEvent]] =
+    processCommandMetric.timeFuture {
+      if (!validIds(aggregate, surgeCommand)) {
+        Future.failed(new Exception("Wrong aggregate ids!"))
+      } else {
+        logger.info(
+          s"Calling command handler of business app via gRPC. Aggregate id: ${surgeCommand.aggregateId})." +
+            s" State defined: ${aggregate.isDefined}. Command payload size: ${surgeCommand.payload.length} (bytes).")
+
+        val maybePbState: Option[protobuf.State] = aggregate.map(surgeState => surgeState: protobuf.State)
+        val pbCommand: protobuf.Command = surgeCommand: multilanguage.protobuf.Command
+        val processCommandRequest = ProcessCommandRequest(aggregateId = surgeCommand.aggregateId, maybePbState, Some(pbCommand))
+        val reply: Future[ProcessCommandReply] = bridgeToBusinessApp.processCommand(processCommandRequest)
 
-      reply.flatMap { processCommandReply =>
-        {
-          if (processCommandReply.isSuccess) {
-            logger.info(s"""Called command handler of business app via gRPC. Got back ${processCommandReply.events.size} events!""")
-            Future.successful {
-              processCommandReply.events.map(e => e: SurgeEvent)
+        reply.flatMap { processCommandReply =>
+          {
+            if (processCommandReply.isSuccess) {
+              logger.info(s"""Called command handler of business app via gRPC. Got back ${processCommandReply.events.size} events!""")
+              Future.successful {
+                processCommandReply.events.map(e => e: SurgeEvent)
+              }
+            } else {
+              logger.info(s"""Called command handler of business app via gRPC. Got a rejection message: ${processCommandReply.rejectionMessage}!""")
+              Future.failed(new Exception(processCommandReply.rejectionMessage))
             }
-          } else {
-            logger.info(s"""Called command handler of business app via gRPC. Got a rejection message: ${processCommandReply.rejectionMessage}!""")
-            Future.failed(new Exception(processCommandReply.rejectionMessage))
           }
         }
       }
     }
-  }
 
-  override def handleEvents(aggregate: Option[SurgeState], surgeEvents: Seq[SurgeEvent]): Future[Option[SurgeState]] = {
-    if (surgeEvents.isEmpty) {
-      logger.warning("handleEvents called but no events provided!")
-      Future.successful(aggregate)
-    } else {
-      if (!validIds(aggregate, surgeEvents)) {
-        Future.failed(new Exception("handleEvents called but wrong aggregate ids!"))
+  private val handleEventsMetric: Timer = metric.timer(MetricInfo("HandleEvents", "Handle Events Metric"))
```

Comment:
This name/description for the metric might want to be a bit more descriptive as well.

Reply:
changed.